decidim-proposals/app/helpers/decidim/proposals/proposals_helper.rb

```diff
@@ -20,8 +20,14 @@ def toggle_view_mode_link(current_mode, target_mode, title, params)
         icon_name = target_mode == "grid" ? "layout-grid-fill" : "list-check"
         icon_class = "view-icon--disabled" unless current_mode == target_mode
 
-        link_to path, remote: true, title: do
-          icon(icon_name, class: icon_class, role: "img", "aria-hidden": true)
+        if icon_class == "view-icon--disabled"
+          link_to path, remote: true, role: "button", title: do
+            icon(icon_name, class: icon_class, role: "img", "aria-hidden": true)
+          end
+        else
+          link_to path, remote: true, role: "button", "aria-current": true, title: do
+            icon(icon_name, class: icon_class, role: "img", "aria-hidden": true, style: "border:1px solid black")
+          end
```

Comment:
I would have 2 comments here : 
1. I would use `if current_mode == target_mode` and swap the conditions 
2. Since the icon class is `view-icon--disabled`, and is not used in other parts, i would add the styling into the class itself.